metricbeat/module/system/filesystem/filesystem.go

```diff
@@ -0,0 +1,56 @@
+package filesystem
+
+import (
+	"github.com/elastic/beats/metricbeat/helper"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/topbeat/system"
+)
+
+func init() {
+	helper.Registry.AddMetricSeter("system", "filesystem", New)
+}
+
+// New creates new instance of MetricSeter
+func New() helper.MetricSeter {
+	return &MetricSeter{}
+}
+
+type MetricSeter struct{}
+
+func (m *MetricSeter) Setup(ms *helper.MetricSet) error {
+	return nil
+}
+
+func (m *MetricSeter) Fetch(ms *helper.MetricSet, host string) (event common.MapStr, err error) {
+
+	fss, err := system.GetFileSystemList()
+	if err != nil {
+		logp.Warn("Getting filesystem list: %v", err)
+		return nil, err
+	}
+
+	event = common.MapStr{}
+
+	for _, fs := range fss {
+		fsStat, err := system.GetFileSystemStat(fs)
+		if err != nil {
+			logp.Debug("topbeat", "Skip filesystem %d: %v", fsStat, err)
+			continue
+		}
+		system.AddFileSystemUsedPercentage(fsStat)
+
+		fsEvent := common.MapStr{
+			fsStat.DevName: system.GetFilesystemEvent(fsStat),
```

Comment:
What do we want to have as a key here? Filesystem names can also have spaces etc.